developerportal/apps/common/middleware.py

```diff
@@ -6,6 +6,21 @@
 from ratelimit.core import is_ratelimited
 from ratelimit.exceptions import Ratelimited
 
+RATELIMIT_GROUP_PUBLIC_REQUESTS = "public_requests"
+RATELIMIT_GROUP_ADMIN_REQUESTS = "admin_requests"
+
+
+def _get_group(request):
+    if request.user and request.user.is_authenticated and request.user.is_staff:
+        return RATELIMIT_GROUP_ADMIN_REQUESTS
+    return RATELIMIT_GROUP_PUBLIC_REQUESTS
+
+
+def _get_appropriate_rate(group):
```

Comment:
nit: wondering if a more appropriate (`UnintendedPunException`) would be: `_get_ratelimit_group` for the first one, and `_get_ratelimit_rate` for the second one?